timemachine/fe/single_topology.py

```diff
@@ -1835,144 +1750,27 @@ def mol(self, lamb: float, min_bond_k: float = DEFAULT_BOND_IS_PRESENT_K) -> Che
-        omm_topology: OpenMMTopology,
-    ) -> BoundPotential[NonbondedInteractionGroup]:
-        """Parameterize nonbonded interactions between the host and guest"""
-        num_host_atoms = host_nonbonded.params.shape[0]
-        num_guest_atoms = self.get_num_atoms()
-        cutoff = host_nonbonded.potential.cutoff
-
-        guest_ixn_env_params = self._get_guest_params(self.ff.q_handle, self.ff.lj_handle, lamb, cutoff)
-
-        # L-W terms
-        num_other_atoms = num_host_atoms - num_water_atoms
-
-        def get_lig_idxs() -> NDArray[np.int32]:
-            return np.arange(num_guest_atoms, dtype=np.int32) + num_host_atoms
-
-        def get_water_idxs() -> NDArray[np.int32]:
-            return np.arange(num_water_atoms, dtype=np.int32) + num_other_atoms
-
-        def get_other_idxs() -> NDArray[np.int32]:
-            return np.arange(num_other_atoms, dtype=np.int32)
-
-        def get_env_idxs() -> NDArray[np.int32]:
-            return np.concatenate([get_other_idxs(), get_water_idxs()])
-
-        hg_nb_ixn_params = host_nonbonded.params.copy()
-        if ff.env_bcc_handle is not None:
-            env_bcc_h = ff.env_bcc_handle.get_env_handle(omm_topology, ff)
-            hg_nb_ixn_params[:, NBParamIdx.Q_IDX] = env_bcc_h.parameterize(ff.env_bcc_handle.params)
-
-        ixn_pot, ixn_params = get_ligand_ixn_pots_params(
-            get_lig_idxs(),
-            get_env_idxs(),
-            hg_nb_ixn_params,
-            guest_ixn_env_params,
-            beta=host_nonbonded.potential.beta,
-            cutoff=cutoff,
-        )
+    def _wrap_align_host_nonbonded(self, host_nonbonded: BoundPotential[Nonbonded]) -> AlignedNonbondedAllPairs:
+        lhs_nb_ap = build_end_state_reference_all_pairs(self.mol_a.GetNumAtoms(), host_nonbonded)
+        rhs_nb_ap = build_end_state_reference_all_pairs(self.mol_b.GetNumAtoms(), host_nonbonded)
+        camm = build_combined_atom_map_mixin(host_nonbonded.potential.num_atoms, self.mol_a, self.mol_b, self.core)
+        return self._align_nonbonded_all_pairs(lhs_nb_ap, rhs_nb_ap, camm)
 
-        bound_ixn_pot = ixn_pot.bind(ixn_params)
-        return bound_ixn_pot
+    def _wrap_aligned_host_guest_nonbonded_ixn_group(
+        self, host_nonbonded: BoundPotential[Nonbonded]
+    ) -> AlignedNonbondedInteractionGroup:
+        lhs_nb_ixn_group = build_end_state_reference_ixn_group(self.mol_a, host_nonbonded, self.ff)
+        rhs_nb_ixn_group = build_end_state_reference_ixn_group(self.mol_b, host_nonbonded, self.ff)
+        camm = build_combined_atom_map_mixin(host_nonbonded.potential.num_atoms, self.mol_a, self.mol_b, self.core)
+        return self._align_nonbonded_interaction_group(lhs_nb_ixn_group, rhs_nb_ixn_group, camm)
 
     def combine_with_host(
         self,
         host_system: HostSystem,
         lamb: float,
-        num_water_atoms: int,
-        ff: Forcefield,
-        omm_topology: OpenMMTopology,
+        num_water_atoms: int,  # deprecated/unused
```

Comment:
This will be removed in a future PR.

Reply:
(comment is wrong - these still need to be here for now since the custom interaction group needs them.)